plugin/storage/es/options.go

```diff
@@ -232,6 +232,12 @@ func addFlags(flagSet *flag.FlagSet, nsConfig *namespaceConfig) {
 		"Use read and write aliases for indices. Use this option with Elasticsearch rollover "+
 			"API. It requires an external component to create aliases before startup and then performing its management. "+
 			"Note that "+nsConfig.namespace+suffixMaxSpanAge+" will influence trace search window start times.")
+	flagSet.Bool(
```

Comment:
Q: how is UseILM different from UseReadWriteAliases option above?

Reply:
We can still use write and read aliases and manage the indices ourselves using esRollover and esCleaner scripts without ILM ( which has been the case until now). We wanted to be backward compatible and enable users to ramp up to ILM or go back to managing indices manually whenever they want. 

@albertteoh  Kindly add if I missed anything.

Reply:
`UseReadWriteAliases` tells Jaeger components to read and write from a single read and write index. As @bhiravabhatla mentioned, users may then implement their own manual index rollover and cleanup.

`UseILM` tells Elasticsearch to manage the rollover and cleanup (within the index mapping) and **requires** index aliases to be used. IIUC, if aliases aren't used, ILM just does not work.

Given that, would it make sense to either return an error if `UseILM == true && UseReadWriteAliases == false`? We may have discussed this in the past.

Reply:
Yes thats correct, ILM doesnot work without aliases. And it does make sense to add above validation. Will do in next commit

Reply:
Is it ok if I open another issue to add this check? @albertteoh

Reply:
Yes, that's fine @bhiravabhatla.

Reply:
@albertteoh - Want to create a followup issue for this. Could you help me with where (which package) can be used to add this validation & should we panic - if validation fails?

Reply:
#2814 Let me know if that proposal makes sense.